### ISSUE TITLE
next start does not stop next-server on kill signal

### ISSUE THREAD

**Issue by strowk** [vercel/next.js#64778]
### Link to the code that reproduces this issue

https://github.com/strowk/nextjs-leak

### To Reproduce

0. Use linux and bash
1. git clone https://github.com/strowk/nextjs-leak && cd nextjs-leak
2. run `yarn && yarn build`
3. run `yarn start &`
4. save pid of yarn process `server_pid=$!`
5. check processes running `ps aux | grep next` (there would be `next start` and `next-server`)
6. run `kill $server_pid` to stop the server
7. run `ps aux | grep next` to check what is still running

### Current vs. Expected behavior

Expected: no next processes running
Current: `next start` does not run, but `next-server` is still running

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #111~20.04.1-Ubuntu SMP Mon Mar 11 15:44:43 UTC 2024
Binaries:
  Node: 20.5.0
  npm: 9.8.0
  Yarn: 1.22.21
  pnpm: N/A
Relevant Packages:
  next: 14.0.2
  eslint-config-next: 14.0.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next start (local)

### Additional context

This is not reproducing on MacOS, where the difference is that `next start` process does not exist and yarn appears to pass kill signal directly to next-server (or so I assume).

**Comment by boyland** [comment]
I'm having the problem too in Ubuntu 24.04: If I start the development server using "npm run dev &" and then try to kill it by killing the process, it leaves node and next-server running.

**Comment by Tymek** [comment]
yup, `next start` somehow spawns another process with another PID as server 
for some reason on my machine `lsof -ti:3033` didn't show it, `ss -ltnp | grep ':3033'` did

**Comment by UnusualFrog** [comment]
Having same issue on Linux Mint 22.2; Killing the "npm run dev" parent process fixed the issue

**Comment by marcoshernanz** [comment]
Thanks for the report. I reproduced this on Ubuntu, but the signal stops at Yarn's shell child and never reaches Next.js. When Next.js receives the signal directly, it exits and closes the port, and a plain Node server produces the same orphan behind this shell chain. This is package-runner behavior rather than a Next.js bug. For supervised deployments, run Next.js or the standalone Node entrypoint directly. Closing as external to Next.js.
